### PR TITLE
orchestrator proto refactor Phase 1 - GetRuntime

### PR DESCRIPTION
**Pull request by hoozecn**
#### What type of this PR

Add one of the following kinds:
/kind polish

#### Which issue(s) this PR fixes:

- [「重构」orchestrator runtime 部分改为使用proto定义](https://erda-org.erda.cloud/erda/dop/projects/387/issues/all?id=217750&issueFilter__urlQuery=eyJzdGF0ZUJlbG9uZ3MiOlsiT1BFTiIsIldPUktJTkciLCJXT05URklYIiwiUkVPUEVOIl0sImFzc2lnbmVlSURzIjpbIjEwMDEzNTYiXX0%3D&issueTable__urlQuery=eyJwYWdlTm8iOjF9&issueViewGroup__urlQuery=eyJ2YWx1ZSI6ImthbmJhbiIsImNoaWxkcmVuVmFsdWUiOnsia2FuYmFuIjoiZGVhZGxpbmUifX0%3D&iterationID=506&type=TASK)

#### Response compare

Left if after, Right is before
```
Text Compare
Produced: 2021/9/3 15:52:16

Mode:  All, With Context
File: 
 1 {                                                                    =   1 {
 2   "success": true,                                                       2   "success": true,
 3   "data": {                                                              3   "data": {
 4     "id": 1,                                                             4     "id": 1,
 5     "name": "master",                                                    5     "name": "master",
 6     "serviceGroupName": "56e0097541",                                    6     "serviceGroupName": "56e0097541",
 7     "serviceGroupNamespace": "services",                                 7     "serviceGroupNamespace": "services",
 8     "source": "PIPELINE",                                                8     "source": "PIPELINE",
 9     "status": "Healthy",                                                 9     "status": "Healthy",
10     "deployStatus": "OK",                                               10     "deployStatus": "OK",
11     "deleteStatus": "",                                                 11     "deleteStatus": "",
12     "releaseId": "40ca218818e140b78df8a72cb39d26b9",                    12     "releaseId": "40ca218818e140b78df8a72cb39d26b9",
13     "clusterId": 1,                                                     13     "clusterId": 1,
14     "clusterName": "terminus-dev",                                      14     "clusterName": "terminus-dev",
15     "clusterType": "k8s",                                               15     "clusterType": "k8s",
16     "resources": {                                                      16     "resources": {
17       "cpu": 0.4,                                                       17       "cpu": 0.4,
18       "mem": 40,                                                        18       "mem": 40,
19       "disk": 0                                                         19       "disk": 0
20     },                                                                  20     },
21     "extra": {                                                          21     "extra": {
22       "applicationId": 2,                                               22       "applicationId": 2,
23       "buildId": 17659,                                                 23       "buildId": 17659,
24       "workspace": "PROD"                                               24       "workspace": "PROD"
25     },                                                                  25     },
26     "projectID": 2,                                                     26     "projectID": 2,
27     "services": {                                                       27     "services": {
28       "go-demo": {                                                      28       "go-demo": {
29         "status": "Healthy",                                            29         "status": "Healthy",
30         "deployments": {                                                30         "deployments": {
31           "replicas": 2                                                 31           "replicas": 2
32         },                                                              32         },
33         "resources": {                                                  33         "resources": {
34           "cpu": 0.2,                                                   34           "cpu": 0.2,
35           "mem": 20,                                                    35           "mem": 20,
36           "disk": 0                                                     36           "disk": 0
37         },                                                              37         },
38         "envs": {                                                       38         "envs": {
39           "ENV_A": "A"                                                  39           "ENV_A": "A"
40         },                                                              40         },
41         "addrs": [                                                      41         "addrs": [
42           "go-demo-56e0097541.project-2-prod.svc.cluster.local:8080"    42           "go-demo-56e0097541.project-2-prod.svc.cluster.local:8080"
43         ],                                                              43         ],
44         "expose": [                                                     44         "expose": [
45           "http://go-demo-prod-1-app.dev.terminus.io"                   45           "http://go-demo-prod-1-app.dev.terminus.io"
46         ],                                                              46         ],
--------------------------------------------------------------------------------------------------------------------------------------------------
47         "errors": []                                                 <> 47         "errors": null
--------------------------------------------------------------------------------------------------------------------------------------------------
48       }                                                              =  48       }
49     },                                                                  49     },
50     "lastMessage": {},                                                  50     "lastMessage": {},
--------------------------------------------------------------------------------------------------------------------------------------------------
51     "timeCreated": "2021-08-23T06:44:53Z",                           <> 51     "timeCreated": "2021-08-23T14:44:53+08:00",
52     "createdAt": "2021-08-23T06:44:53Z",                                52     "createdAt": "2021-08-23T14:44:53+08:00",
53     "updatedAt": "2021-09-03T04:10:10Z",                                53     "updatedAt": "2021-09-03T12:10:10+08:00",
54     "errors": []                                                        54     "errors": null
                                                                           55   },
                                                                           56   "err": {
                                                                           57     "code": "",
                                                                           58     "msg": "",
                                                                           59     "ctx": null
--------------------------------------------------------------------------------------------------------------------------------------------------
55   }                                                                  =  60   }
56 }                                                                       61 }
--------------------------------------------------------------------------------------------------------------------------------------------------

```

### CODE REVIEW

File: modules/orchestrator/initialize.go [hunk 1]
```diff
@@ -35,59 +37,40 @@ import (
...
 	// init endpoints
-	ep, err := initEndpoints(p.db)
+	ep, err := initEndpoints(db)
 	if err != nil {
 		return err
 	}
 
 	bdl := bundle.New()
-	server := httpserver.New(conf.ListenAddr())
+	server := httpserver.New(":0")
```
Comment (sfwn) [inline modules/orchestrator/initialize.go:68]: ":0" is strage.

File: modules/orchestrator/initialize.go [hunk 2]
```diff
@@ -35,59 +37,40 @@ import (
...
 	server.WithLocaleLoader(bdl.GetLocaleLoader())
 	// server.Router().Path("/metrics").Methods(http.MethodGet).Handler(promxp.Handler("orchestrator"))
 	server.RegisterEndpoint(ep.Routes())
-	p.server = server
+
+	ctx.Service("http-server").(infrahttpserver.Router).Any("/**", server.Router())
```
Comment (sfwn) [inline modules/orchestrator/initialize.go:73]: cool

**Review by sfwn — CHANGES_REQUESTED** [review]

File: modules/orchestrator/initialize.go (hunk 1 above)
Comment (hoozecn) [reply modules/orchestrator/initialize.go:68]: This server will never be started @sfwn

Comment (hoozecn) [reply modules/orchestrator/initialize.go:69]: :0 will listen on a random port

**Review by sfwn — APPROVED** [review]

**Pull request merged**